src/rooms.js

```diff
@@ -0,0 +1,11 @@
+function findAvailableRoomsByDate(date, bookings, rooms) {
+  const alreadyBookedRoomsOnDate = bookings.filter(booking => booking.date === date);
+  const alreadyBookedRoomNumbers = alreadyBookedRoomsOnDate.map(booking => booking.roomNumber);
+  const availableRooms = rooms.filter(room => !alreadyBookedRoomNumbers.includes(room.number));
```

Comment:
Similarly to the comment above, you could return rooms.filter on line 4 and delete line 5